libcontainer/specconv/spec_linux.go

```diff
@@ -1016,14 +1016,19 @@ func SetupSeccomp(config *specs.LinuxSeccomp) (*configs.Seccomp, error) {
 		return nil, nil
 	}
 
-	// We don't currently support seccomp flags.
-	if len(config.Flags) != 0 {
-		return nil, errors.New("seccomp flags are not yet supported by runc")
-	}
-
 	newConfig := new(configs.Seccomp)
 	newConfig.Syscalls = []*configs.Syscall{}
 
+	// We don't currently support all seccomp flags.
+	for _, flag := range config.Flags {
+		switch flag {
+		case "SECCOMP_FILTER_FLAG_LOG":
```

Comment:
It would be nice to not have this set of flags handled in two separate places. Perhaps we can add some common part to `libcontainer/seccomp`?

While at it, adding `SCMP_FLTATR_CTL_OPTIMIZE` would be awesome

Reply:
Right now we have four supported flags following the same naming convention supported by the SECCOMP_SET_MODE_FILTER operation.

SECCOMP_FILTER_FLAG_LOG
SECCOMP_FILTER_FLAG_NEW_LISTENER
SECCOMP_FILTER_FLAG_SPEC_ALLOW
SECCOMP_FILTER_FLAG_TSYNC

How would `SCMP_FLTATR_CTL_OPTIMIZE` map to that?

Reply:
> While at it, adding `SCMP_FLTATR_CTL_OPTIMIZE` would be awesome

The [set of flags listed in the runtime-spec](https://github.com/opencontainers/runtime-spec/blob/8958f93039ab90be53d803cd7e231a775f644451/config-linux.md) is a [subset](https://github.com/opencontainers/runtime-spec/pull/1096) of the flag supported by the Linux API in the seccomp syscall:

> flags (array of strings, OPTIONAL) - list of flags to use with seccomp(2).
>
> A valid list of constants is shown below.
>
> SECCOMP_FILTER_FLAG_TSYNC
> SECCOMP_FILTER_FLAG_LOG
> SECCOMP_FILTER_FLAG_SPEC_ALLOW

`SCMP_FLTATR_CTL_OPTIMIZE` is part of libseccomp API but not part of the Linux API. It is not in the runtime-spec because container runtimes don't have to use libseccomp. Are you suggesting runc should support it even if it's not in the spec?

Reply:
> It would be nice to not have this set of flags handled in two separate places. Perhaps we can add some common part to libcontainer/seccomp?

Now that I have added the 2 additional flags, the code looks different, and I don't see how to add some common part to libcontainer/seccomp... any suggestions?

Reply:
Re `SCMP_FLTATR_CTL_OPTIMIZE`.

Perhaps runc should just set it to `2` unconditionally. Also, this is orthogonal to this PR, so nevermind.